src/NFT/AgentCommunication.sol

```diff
@@ -139,4 +139,15 @@ contract AgentCommunication is Ownable {
         emit LogMessage(message.sender, msg.sender, message.message, message.value);
         return message;
     }
+
+    function purgeMessages(address agentAddress) public onlyOwner {
+        delete queues[agentAddress];
+    }
+
+    function purgeAllMessages() public onlyOwner {
+        address[] memory agents = agentRegistry.getAllRegisteredAgents();
+        for (uint256 i = 0; i < agents.length; i++) {
+            purgeMessages(agents[i]);
+        }
+    }
```

Comment:
GPT tells me we can be more gas efficient using the following
```
function purgeAllMessages() public onlyOwner {
    // Create a new mapping instead of iterating and deleting each entry
    delete queues;

    emit AllMessagesPurged(agentRegistry.getAllRegisteredAgents().length);
}
```